src/components/MapBox.tsx

```diff
@@ -273,26 +313,31 @@ export const MapBox: React.FC = () => {
         {eventMarkers}
         {accommodations && accommodations.length > 0
           ? accommodations.map((f) => {
+              const isSelected = f.id === selectedFacilityId;
               if (f.location && f.location.coordinates) {
                 return (
                   <Marker
                     key={f.id}
-                    icon={getPriceMarkerIcon(f.lowestPrice)}
+                    icon={getPriceMarkerIcon(f.lowestPrice, isSelected)}
                     position={[f.location.coordinates[1], f.location.coordinates[0]]}
                     eventHandlers={{
-                      click: () => selectFacility(f.id)
+                      click: () =>
+                        isSelected ? unSelectFacility() : selectFacility(f.id)
                     }}
                   >
-                    <Popup>
-                      <SearchCard
-                        key={f.id}
-                        sm={true}
-                        facility={f}
-                        isSelected={f.id === selectedFacilityId}
-                        numberOfDays={numberOfDays}
-                        focusedEvent={f.eventInfo}
-                      />
-                    </Popup>
+                    {isMobileView ? null : (
```

Comment:
in the `SearchForm` it is called `isMobile` and here it is called `isMobileView` 

to keep it consistent, i suggest to pick one name.